### PR TITLE
irace 3.4.1 (new formula)

### PR DESCRIPTION
**Pull request by DE0CH**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/irace.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    resource("R6").stage do
+      system "R", "CMD", "INSTALL", ".", "--library=#{buildpath}/r-lib"
+    end
+    system "R", "CMD", "INSTALL", ".", "--library=#{buildpath}/r-lib"
+    prefix.install Dir["#{buildpath}/r-lib/"]
+    bin.install "#{prefix}/r-lib/irace/bin/irace"
+    bin.install "#{prefix}/r-lib/irace/bin/irace2pyimp"
+    bin.install "#{prefix}/r-lib/irace/bin/parallel-irace"
+    bin.install "#{prefix}/r-lib/irace/bin/parallel-irace-mpi"
+    bin.install "#{prefix}/r-lib/irace/bin/parallel-irace-qsub"
```
Comment (SMillerDev) [inline Formula/irace.rb:26]: Won't this work?
```suggestion
    bin.install Dir["#{buildpath}/r-lib/bin/*"]
    prefix.install Dir["#{buildpath}/r-lib/"]
```

File: Formula/irace.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  def caveats
+    <<~EOS
+      You need to let R know the installation of the irace package by setting an environment variable (and add it to your ~/.profile, ~/.bashrc or ~/.zshrc for it to persist)
+        $ export R_LIBS="#{opt_prefix}/r-lib:${R_LIBS}"
+    EOS
+  end
+
+  test do
+    system "R_LIBS=#{opt_prefix}/r-lib:${R_LIBS} #{bin}/irace -v"
```
Comment (SMillerDev) [inline Formula/irace.rb:37]:
```suggestion
    ENV["R_LIBS"] = opt_prefix}/"r-lib"
    system bin/"irace", "-v"
```

Additionally we need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

**Comment by DE0CH** [comment]
Thanks a lot for the feedback. I have just change the formula. Could you please review it? Thanks.

**Review by SMillerDev — DISMISSED** [review]
I think this is fine, but I'd like to tag in some maintainers with previous comments on R related things

**Comment by DE0CH** [comment]
That would be great. Thanks.

**Comment by Bo98** [comment]
I'll defer to Jon since he knows more about this than me.

My Python background does pose the question whether it would make sense to install directly into R's site-library instead so it works out-of-the-box without envs but if users can just do `R cmd remove` then perhaps it's a mess worth avoiding.

**Comment by Bo98** [comment]
> homebrew does not have premission to install it directly into the R's site-library.

Yes, installing outside the formula's Cellar is forbidden.

What you would do is install in `#{lib}/R/#{Formula["r"].version.major_minor}/site-library` instead. When the formula is installed it should symlink into the correct place.

**Comment by DE0CH** [comment]
Thanks I just changed the formla. Please review.

**Comment by DE0CH** [comment]
Thanks. It works perfectly. I just chagned the formula. Please review.

**Comment by DE0CH** [comment]
Sorry @[SMillerDev](https://github.com/SMillerDev), didn't mean to request review from you. I thought the request review button would only send Bo98 and jonchang a notification... 😅

**Comment by DE0CH** [comment]
Sorry @[SMillerDev](https://github.com/SMillerDev), didn't mean to request review from you. I thought the request review button would only send Bo98 and jonchang a notification... 😅

**Comment by jonchang** [comment]
I'm going to turn this down, partly because the GitHub repository for irace doesn't meet our notability requirements, and partly because having `depends_on "r"` is really unpleasant for users, especially on macOS.

Many users will also have the cask version installed (though likely not from `brew install --cask r`, just from downloading it from the R website) and there's a serious possibility of frankeninstalls because if you've installed R from either the cask or the website download, Homebrew will refuse to link the brewed version of R, and it'll be unclear what copy of R will be running irace. Brewed R also has the problem where you don't get any CRAN binaries. 

In short, there's a big mismatch between how the R package ecosystem works and how Homebrew packages apps. If you still want to package this for Homebrew, I suggest [creating your own tap](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap) and using a [custom Requirement](https://rubydoc.brew.sh/Requirement.html) like I have in [this (somewhat outdated) formula](https://github.com/jonchang/homebrew-biology/blob/aba7560d286fb9231e18634b4ed855aa3b9cb1dc/Formula/sowhat.rb).
